app/src/main/res/values/strings.xml

```diff
@@ -2,16 +2,16 @@
 <resources>
 
     <!-- Applications list -->
-    <string name="info_applications_list_refreshed">Applications list refreshed</string>
-    <string name="button_refresh_list">Refresh the applications list</string>
-    <string name="error_application_not_found">The application %1$s has not been found!</string>

```

Comment:
Out of reference name, exclamation mark for no good reason.

Reply:
> Out of reference name

Do you mean the name shouldn't be part of this string? (ex: "Could not find the app.")

> exclamation mark for no good reason.
This is in the style of an early 90s error message.

Hum, I could probably find software more recent than early 90s and still using exclamation marks for error messages.
But indeed we can remove it if you think this is not appropriate here.

Since this error message is supposed to be displayed in a temporary bubble (what Android names "toasts"), maybe we could even drop the dot...

Thinking about it, maybe `Could not find \"%1$s\"` would be even simpler.
For example, if the not found application is named Contacts, the message would be: `Could not find "Contacts"`

Reply:
Yes.
It isn't just the exclamation, it is the passive form with it.
It is a garden-path sentence, and sort of robotic.

If at all possible I like to go "Install a contact list app first" to make it direct. Then again I don't know if it fits the bill here, which is sort of the problem to solve.

Reply:
> If at all possible I like to go "Install a contact list app first" to make it direct. Then again I don't know if it fits the bill here

Not really, this message is displayed when the user clicks on an application from the list and it cannot be started (the technical reason behind it is that the Android "Intent" object needed to launch the app has not been found).
This happens only in some corner cases, the most frequent being probably the disabling of a pre-installed app.

When the user add or remove an app, the Android system sends a broadcast message that we can receive to update the app list automatically. However, there is no such message when the user disable an app that cannot be uninstalled so the list cannot be refreshed accordingly.
It can be updated manually by the user or later with some events (app added/removed, launcher restarted, icon pack changed) but meanwhile the icon of the disabled app is still visible and clickable.
=> When clicking on it, the app is not found and the message is displayed with "%1$s" replaced by the app name.

Reply:
Thinking about it, with "Could not find %1$s" which is quite generic and flexible, maybe I could use this message for other places in the code and remove some other strings saying more or less the same things (I think about some for the shortcuts, or the icon pack loading).

Reply:
Saves time for translators.
However, it fails if the verb is gendered wrt. the noun.
I don't really know, will have to see.

Reply:
Turns out having the full sentence is a feature that avoids this problem.
Better to keep as is.